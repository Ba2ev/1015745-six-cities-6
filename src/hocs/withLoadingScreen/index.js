import withLoadingScreen from './with-loading-screen';
export default withLoadingScreen;
