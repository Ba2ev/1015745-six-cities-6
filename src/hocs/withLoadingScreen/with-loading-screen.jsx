import React from 'react';
import PropTypes from 'prop-types';
import LoadingScreen from '../../components/loading-screen';

const withLoadingScreen = (Component) => {
  const WithLoadingScreen = (props) => {
    return props.isLoaded ? <Component {...props}/> : <LoadingScreen />;
  };

  WithLoadingScreen.propTypes = {
    isLoaded: PropTypes.bool.isRequired,
  };

  return WithLoadingScreen;
};

export default withLoadingScreen;
