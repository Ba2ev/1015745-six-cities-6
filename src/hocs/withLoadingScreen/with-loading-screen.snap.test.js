import React from 'react';
import {render} from '@testing-library/react';
import withLoadingScreen from './with-loading-screen';

describe(`should HOC withLoadingScreen render correctly`, () => {
  it(`should HOC withLoadingScreen render correctly after loaded data`, () => {
    const Component = () => <h1>Intro Component</h1>;
    const ComponentWrapped = withLoadingScreen(Component);

    const {container} = render(<ComponentWrapped isLoaded/>);

    expect(container).toMatchSnapshot();
  });

  it(`should HOC withLoadingScreen render correctly with loading data`, () => {
    const Component = () => <h1>Intro Component</h1>;
    const ComponentWrapped = withLoadingScreen(Component);

    const {container} = render(<ComponentWrapped isLoaded={false}/>);

    expect(container).toMatchSnapshot();
  });
});
