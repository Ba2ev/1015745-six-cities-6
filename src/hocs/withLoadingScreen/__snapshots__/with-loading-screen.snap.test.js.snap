// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HOC withLoadingScreen render correctly should HOC withLoadingScreen render correctly after loaded data 1`] = `
<div>
  <h1>
    Intro Component
  </h1>
</div>
`;

exports[`should HOC withLoadingScreen render correctly should HOC withLoadingScreen render correctly with loading data 1`] = `
<div>
  <p>
    Loading ...
  </p>
</div>
`;
